src/libraries/System.Private.Uri/src/System/ValueStringBuilderExtensions.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Internal.Runtime.CompilerServices;
+
+namespace System.Text
+{
+    internal ref partial struct ValueStringBuilder
+    {
+        public void AppendString(string value) => AppendSlow(value);
+
+        public void Append(string source, int offset, int length) => Append(source.AsSpan(offset, length));
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Append(char c1, char c2)
```

Comment:
This is a lot of code. Is it really that much faster than calling `Append(char)` twice?

Reply:
Not in this specific case since I did not change other call sites that could benefit. Reverted as out of scope for this PR (https://github.com/dotnet/runtime/pull/34864#discussion_r407228866).